validator/keymanager/remote-web3signer/client.go

```diff
@@ -0,0 +1,177 @@
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp.Body)
+	jsonDataFromHttp, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read response body")
+	}
+	signResp := &signResponse{}
+	if err := json.Unmarshal(jsonDataFromHttp, signResp); err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, failed to unmarshal json response")
+	}
+	decoded, err := decodeHex(signResp.Signature)
+	if err != nil {
+		return nil, err
+	}
+	blsSig, err := bls.SignatureFromBytes([]byte(decoded))
+	if err != nil {
+		return nil, err
+	}
+	return blsSig, nil
+}
+
+func (client *client) GetPublicKeys() ([][]byte, error) {
+	const requestPath = "/publicKeys"
+	resp, err := client.doRequest(http.MethodGet, client.BasePath+requestPath, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp.Body)
+	var publicKeys []string
+	if err := json.NewDecoder(resp.Body).Decode(&publicKeys); err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, unable to read response body as array of strings")
+	}
+	decodedKeys := make([][]byte, len(publicKeys))
+	var errorMessage string
+	for i, value := range publicKeys {
+		decodedKey, err := decodeHex(value)
+		if err != nil {
+			if errorMessage == "" {
+				errorMessage = "Failed to decode from Hex from the following public keys: "
+			}
+			errorMessage += value + " "
+			continue
+		}
+		decodedKeys[i] = decodedKey[:]
+	}
+	if errorMessage != "" {
+		return nil, errors.New(errorMessage)
+	}
+	return decodedKeys, nil
+}
+
+func (client *client) ReloadSignerKeys() error {
+	const requestPath = "/reload"
+	resp, err := client.doRequest(http.MethodPost, client.BasePath+requestPath, nil)
+	if err != nil {
+		return err
+	}
+	defer closeBody(resp.Body)
+	return nil
+}
+
+func (client *client) GetServerStatus() (string, error) {
+	const requestPath = "/upcheck"
+	resp, err := client.doRequest(http.MethodGet, client.BasePath+requestPath, nil)
+	if err != nil {
+		return "", err
+	}
+	defer closeBody(resp.Body)
+	var status string
+	if err := json.NewDecoder(resp.Body).Decode(&status); err != nil {
+		return "", errors.Wrap(err, "Invalid Format, unable to read response body as a string")
+	}
+	return status, nil
+}
+
+func (client *client) doRequest(httpMethod string, fullPath string, body io.Reader) (*http.Response, error) {
+	req, err := http.NewRequest(httpMethod, fullPath, body)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid format, failed to create new Post Request Object")
+	}
+	resp, err := client.restClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to execute json request")
+	}
+	return resp, nil
+}
+
+func decodeHex(signature string) ([]byte, error) {
+	decoded, err := hex.DecodeString(strings.TrimPrefix(signature, "0x"))
+	if err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, failed to unmarshal json response")
+	}
+	return decoded, nil
+}
+func closeBody(body io.ReadCloser) {
```

Comment:
```suggestion
func closeBody(body io.Closer) {
```
io Closer is enough here